### PR TITLE
META-3091 implemented pagination on classification deletion task

### PR DESCRIPTION
**Pull request by BurakAltas**
## Paginate commit on classification propagation delete task

> With [META-3047](https://linear.app/atlanproduct/issue/META-3047/optimize-classification-propagation-addremoval task-to-reduce-time), graph commit is called in chunks and this affected performance, heap usage, success of the ADD task drastically.  So, the same logic has to be applied to the delete task as well.

## Type of change
- [x] Enhancement

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java [hunk 1]
```diff
@@ -3298,20 +3281,51 @@ public List<String> deleteClassificationPropagation(String entityGuid, String cl
...
 
-            List<AtlasEntity>   propagatedEntities = updateClassificationText(classification, entityVertices);
+            do {
+                toIndex = ((offset + CHUNK_SIZE > propagatedEdgesSize) ? (int) propagatedEdgesSize : (offset + CHUNK_SIZE));
 
-            entityChangeNotifier.onClassificationsDeletedFromEntities(propagatedEntities, Collections.singletonList(classification));
+                List<AtlasVertex> entityVertices = deleteDelegate.getHandler().removeTagPropagation(classification, propagatedEdges.subList(offset, toIndex));
+
+                if (CollectionUtils.isEmpty(entityVertices)) {
+                    return null;
```
Comment (DillionTan) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:3305]: don't we need to continue for rest of chunks although intermediate chunks have entity vertices that have already been removed?

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java [hunk 2]
```diff
@@ -3298,20 +3281,51 @@ public List<String> deleteClassificationPropagation(String entityGuid, String cl
...
+            while (offset < propagatedEdgesSize);
+
+            deleteDelegate.getHandler().deleteClassificationVertex(classificationVertex, true);
+
+            transactionInterceptHelper.intercept();
 
-            return propagatedEntities.stream().map(x -> x.getGuid()).collect(Collectors.toList());
+            return deletedPropagationsGuid;
         } catch (Exception e) {
             throw new AtlasBaseException(e);
```
Comment (DillionTan) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:3330]: can we add error log here?

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java (hunk 1 above)
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:3305]: Yes we need to continue here. Have to fix it

Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:3305]: Line:3304 if (CollectionUtils.isEmpty(entityVertices)) {...

I think no need for this check given that propagatedEdges will never be empty at this point & so deleteDelegate.getHandler().removeTagPropagation(..) must return non-empty list

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java (hunk 2 above)
Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:3330]: +1, let us add LOG.error here

**Review by nikhilbonte21 — CHANGES_REQUESTED** [review]

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java (hunk 1 above)
Comment (BurakAltas) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:3305]: @sumandas0  are you fixing it or should I update?

Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java]: Yes I've fixed that

**Pull request merged**